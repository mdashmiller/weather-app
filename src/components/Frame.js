import styled from "styled-components"

const Frame = styled.div`
	width: 300px;
	margin: 40px auto 30px;
	text-align: center;
	font-family: 'Varela Round', Arial, sans-serif;
`

export default Frame
