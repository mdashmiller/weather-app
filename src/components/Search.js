import styled, { css } from "styled-components"

const Search = styled.input`
	background: #eee;
	border: 1px solid #c3c3c3;
	border-radius: 10px;
	margin: 200px auto 20px;
	width: 200px;
	height: 34px;
	padding-left: 8px;
	font-family: 'Font Awesome 5 Free', 'Varela Round', Arial, sans-serif;
	font-weight: 900;
	font-size: 18px;

	&::placeholder {
		color: #6a6a6a;
	}

	${props => props.short && css`
	    width: 114px;
	    padding-left: 5px;
	    margin: 0;
	  `}

	${props => props.landingPage && css`
		display: block;
	    width: 172px;
	   	margin: 26px auto;
		text-align: center;
		padding: 0;
		cursor: pointer;  
	  `}

	${props => props.userLocation && css`
		border: 3px solid #f46403;
			
			&:hover {
				background: #f46403;
				
				&::placeholder {
					color: #eee;
				}
			}	
	  `}

	${props => props.userZip && css`
		border: 3px solid #6a6a6a;

			&:hover {
				background: #6a6a6a;
				
				&::placeholder {
					color: #eee;
				}
			} 
	  `}
`

export default Search
