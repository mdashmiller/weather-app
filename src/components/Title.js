import React from 'react'
import { Link } from 'react-router-dom'

const Title = () =>
	<Link to='/'>
		<h1 id="landing-title" className="title-gold">
			Weather
			<span className="title-grey">
				Now
			</span>
		</h1>
	</Link>

export default Title
