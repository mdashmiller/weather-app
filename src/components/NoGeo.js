import React from 'react'
import { Link } from 'react-router-dom'
import Frame from './Frame'
import Search from './Search'

const NoGeo = () =>
	<Frame>
		<Link to='/'>
			<h1 id="landing-title" className="title-gold">
				Weather
				<span className="title-grey">
					Now
				</span>
			</h1>
		</Link>
		<h2>It looks like geolocation is disabled.</h2>
		<h2>Would you like to search for weather by zipcode?</h2>
		<Link to='/lookup-by-zip'>
			<Search
				type="text"
				placeholder="&#xf002; Lookup By Zip"
				noGeo
				lookUp
			/>
		</Link>
	</Frame>

export default NoGeo
