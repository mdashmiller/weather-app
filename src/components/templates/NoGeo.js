import React from 'react'
import { Link } from 'react-router-dom'
import Frame from '../ui/Frame'
import Title from '../ui/Title'
import Search from '../ui/Search'

const NoGeo = () =>
	<Frame>
		<Title />
		<h2>It looks like geolocation is disabled.</h2>
		<h2>Would you like to search for weather by zipcode?</h2>
		<Link to='/lookup-by-zip'>
			<Search
				type="text"
				placeholder="&#xf002; Lookup By Zip"
				noGeo
				lookUp
			/>
		</Link>
	</Frame>

export default NoGeo
