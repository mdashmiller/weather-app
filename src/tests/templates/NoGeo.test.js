import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { BrowserRouter, Link } from 'react-router-dom'

import NoGeo from '../../components/templates/NoGeo'
import Frame from '../../components/ui/Frame'
import Title from '../../components/ui/Title'
import Search from '../../components/ui/Search'

Enzyme.configure({ adapter: new Adapter() })

describe('<NoGeo /> rendering', () => {

	it('renders without crashing', () => {
	  	const div = document.createElement('div')
	  	ReactDOM.render(
	        <BrowserRouter>
	  			<NoGeo />
	  	   	</BrowserRouter>,
	  	div)
	  	ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<BrowserRouter>
				<NoGeo />
		   	</BrowserRouter>
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('renders 1 <Frame> component', () => {
		const wrapper = shallow(<NoGeo />)
		expect(wrapper.find(Frame).length).toBe(1)
	})

	it('renders 1 <Title> component', () => {
		const wrapper = shallow(<NoGeo />)
		expect(wrapper.find(Title).length).toBe(1)
	})

	it('renders 2 h2 elements', () => {
		const wrapper = shallow(<NoGeo />)
		expect(wrapper.find('h2').length).toBe(2)
	})

	it('renders 1 <Link> component', () => {
		const wrapper = shallow(<NoGeo />)
		expect(wrapper.find(Link).length).toBe(1)
	})

	it('renders 1 <Search> component', () => {
		const wrapper = shallow(<NoGeo />)
		expect(wrapper.find(Search).length).toBe(1)
	})	

})

describe('<NoGeo /> Link functionality', () => {

	it('contains a link to get weather by zipcode', () => {
		const wrapper = shallow(<NoGeo />)
		expect(wrapper.find(Link).prop('to')).toBe('/lookup-by-zip')
	})

})
