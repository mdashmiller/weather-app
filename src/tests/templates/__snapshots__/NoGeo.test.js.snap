// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoGeo /> rendering has a valid snapshot 1`] = `
<div
  className="sc-bdVaJa cOdhsb"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="title-gold"
      id="landing-title"
    >
      Weather
      <span
        className="title-grey"
      >
        Now
      </span>
    </h1>
  </a>
  <h2>
    It looks like geolocation is disabled.
  </h2>
  <h2>
    Would you like to search for weather by zipcode?
  </h2>
  <a
    href="/lookup-by-zip"
    onClick={[Function]}
  >
    <input
      className="sc-bwzfXH gJAmOf"
      placeholder="ï€‚ Lookup By Zip"
      type="text"
    />
  </a>
</div>
`;
